### PR TITLE
Support for Default tags

### PR DESCRIPTION
**Pull request by venkat-iblox**

### CODE REVIEW

File: client/configuration.go [hunk 1]
```diff
@@ -18,11 +19,11 @@ const (
 	HeaderClient        = "x-infoblox-client"
 	HeaderSDK           = "x-infoblox-sdk"
 	HeaderAuthorization = "Authorization"
+	version             = "0.1"
+	sdkIdentifier       = "golang-sdk"
+	clientIdentifier    = "automation"
```
Comment (venkat-iblox) [inline client/configuration.go]: @mathewab Do you know if Ansible maintains an infra state locally like Terraform?

File: client/configuration.go [hunk 2]
```diff
@@ -83,12 +87,26 @@ func (c Configuration) internal(basePath string) (*internal.Configuration, error
...
+		ic.AddDefaultTags(c.DefaultTags)
+	}
+
+	// setting up custom tag to identify the client
+	dfTags := make(map[string]string)
+	// Extract client from ClientName string
+	// Format: <client>/version#commit
+	dfTags[clientIdentifier] = strings.Split(c.ClientName, "/")[0]
+	ic.AddDefaultTags(dfTags)
+
```
Comment (venkat-iblox) [inline client/configuration.go]: Should we leave this section disabled by default? We can add the enable flag through Terraform provider configuration

Comment (venkat-iblox) [reply client/configuration.go]: removed this section of the code, as we no longer implement this

File: client/configuration.go [hunk 3]
```diff
@@ -43,6 +42,9 @@ type Configuration struct {
 	// HTTPClient to use for the SDK.
 	// Optional. The default HTTPClient will be used if not provided.
 	HTTPClient *http.Client
+
+	// Default global tags the client can set for all requests.
```
Comment (mathewab) [inline client/configuration.go]:
```suggestion
	// Default tags that will be set in all objects that support tags.
```

File: client/configuration.go [hunk 4]
```diff
@@ -83,12 +85,19 @@ func (c Configuration) internal(basePath string) (*internal.Configuration, error
...
 		Servers:          []internal.ServerConfiguration{{URL: cspURL}},
 		HTTPClient:       httpClient,
-	}, nil
+		DefaultTags:      make(map[string]string),
+	}
+	// Add default tags set
+	if c.DefaultTags != nil {
+		ic.AddDefaultTags(c.DefaultTags)
+	}
+
```
Comment (mathewab) [inline client/configuration.go]: Why do you need this indirection? Can't you just do directly `ic.DefaultTags = c.DefaultTags` ?

Comment (venkat-iblox) [reply client/configuration.go]: Yes, we can. The indirection was added to support `system tag`. I will remove it.

File: internal/configuration_test.go [hunk 5]
```diff
@@ -66,7 +65,7 @@ func TestConfiguration_AddDefaultTags(t *testing.T) {
 			c := &Configuration{
 				DefaultTags: tt.fields.DefaultTags,
 			}
-			c.AddDefaultTags(tt.args.m)
+			c.DefaultTags = tt.args.m
```
Comment (mathewab) [inline internal/configuration_test.go]: can't you remove the whole test?

Comment (venkat-iblox) [reply internal/configuration_test.go]: removed

File: internal/configuration.go [hunk 6]
```diff
@@ -98,6 +100,10 @@ func (c *Configuration) AddDefaultHeader(key string, value string) {
 	c.DefaultHeader[key] = value
 }
 
+func (c *Configuration) GetDefaultTags() map[string]string {
```
Comment (mathewab) [inline internal/configuration.go:103]: We don't usually do `Get...` methods in Go. We can update get generator and fix that in another PR.

**Review by mathewab — APPROVED** [review]

**Pull request merged**